softhddevice.cpp

```diff
@@ -453,6 +464,52 @@ void cSoftHdDevice::OnEnteringState(enum State state) {
 		case STILL_PICTURE:
 			m_pRender->SetDeinterlacerDeactivated(true);
 			break;
+		case DETACHED:
+			m_skipstream = true;
+			// do the same cleanup as in STOP first
+			m_pRender->CancelFilterThread();
+
+			m_pRender->Reset();
+			m_pRender->DestroyFrameBuffers();
+			m_pRender->ScheduleDisplayBlackFrame();
+
+			m_videoReassemblyBuffer.Reset();
+			m_pVideoStream->ClearVdrCoreToDecoderQueue();
+			m_pRender->ClearDecoderToDisplayQueue();
+			m_pVideoStream->CloseDecoder();
+
+			m_audioReassemblyBuffer.Reset();
+
+			// now do the detach
+			m_pVideoStream->DecodingThreadResume();
+			m_pRender->DisplayThreadResume();
+
+			m_pAudio->Exit();
+			m_pRender->Exit(); // render must be stopped before videostream!
+			m_pVideoStream->Exit();
+#ifdef USE_GLES
+			m_pOsdProvider->StopOpenGlThread();
+			SetDisableOglOsd();
+#endif
+			delete m_pAudioDecoder; // includes a Close()
+			delete m_pVideoStream;
+			delete m_pRender;
+			delete m_pAudio;
+			break;
+		case ATTACHED:
+#ifdef USE_GLES
+			SetEnableOglOsd();
+#endif
+			m_pAudio = new cSoftHdAudio(this);
+			m_pAudio->LazyInit();
+			m_pRender = new cVideoRender(this);
+			m_pVideoStream = new cVideoStream(this);
+			m_pAudioDecoder = new cAudioDecoder(m_pAudio);
+			m_pRender->Init(); // starts display thread
+			m_pVideoStream->StartDecoder(new cVideoDecoder(m_pRender->HardwareQuirks())); // starts decoding thread
+			// Audio is init lazily (includes starting thread)
+			m_skipstream = false;
+			break;
```

Comment:
You could move this to the leaving action of DETACHED. Then, the attached state can be removed, or?